ext/standard/head.c

```diff
@@ -363,6 +363,17 @@ PHP_FUNCTION(http_response_code)
 
 	if (response_code)
 	{
+		if (SG(headers_sent) && !SG(request_info).no_headers) {
```

Comment:
I tried and arrived at the same solution, I just didn't have time to send the PR in time, you were faster.

It is the same as:https://github.com/php/php-src/blob/2b15061fbb4d59795adcb9b556ab1ceebe831f17/main/SAPI.c#L671

Perhaps, create a generic function and use it in both places, changing only the beginning of the message, something like that.

I believe that this case will not be replicated much, just in these two places.

Anyway, just a suggestion. It's great :)

Reply:
It is, but the error reporting is different (going through the SAPI vs. the normal interface). I'm not sure where to put it/what the best approach would be because of that.

Reply:
It's probably gonna be more tedious to try to generalise such a small fragment than it is to not generalise this. So I would think the current approach is okay.